GDevelop.js/scripts/generate-dts.mjs

```diff
@@ -0,0 +1,214 @@
+      // Re-read the type since unsigned is an unnecessary prefix for typescript
+      Parser.skipWhitespaces();
+      type = Parser.readUntil(' ');
+    }
+    Parser.skipWhitespaces();
+
+    return { type, optional };
+  }
+
+  static readIdentifier() {
+    let name = '';
+    while (
+      (Parser.cc >= 97 && Parser.cc <= 122) || // [a-z]
+      (Parser.cc >= 65 && Parser.cc <= 90) || // [A-Z]
+      (Parser.cc >= 48 && Parser.cc <= 57) || // [1-9]
+      Parser.c === '_'
+    ) {
+      name += Parser.c;
+      Parser.i++;
+    }
+    return name;
+  }
+
+  static readNumber() {
+    let number = '';
+    while (
+      Parser.cc >= 48 &&
+      Parser.cc <= 57 // [1-9]
+    ) {
+      number += Parser.c;
+      Parser.i++;
+    }
+    return parseInt(number, 10);
+  }
+}
+
+const interfaces = [];
+for (const [a, interfaceName, interfaceCode] of bindingsFile.matchAll(
+  /interface ([a-zA-Z]+) {\r?\n?([^}]*)\r?\n}/gm
+)) {
+  const methods = [];
+
+  Parser.target = interfaceCode;
+  while (!Parser.done) {
+    const { type: returnType, optional: optionalReturn } = Parser.readType();
+    const methodName = Parser.readUntil('(');
+    const isConstructor = returnType === 'void' && methodName === interfaceName;
+
+    /** @type {Array<{name:string, type:string, optional:boolean, defaultValue:(number | typeof none)}>} */
+    const parameters = [];
+    Parser.skipWhitespaces();
+    if (Parser.c !== ')')
+      do {
+        if (Parser.c === ',') Parser.i++;
+        const { type, optional } = Parser.readType();
+        Parser.skipWhitespaces();
+        const name = Parser.readIdentifier();
+        Parser.skipWhitespaces();
+
+        let defaultValue = none;
+        if (Parser.c === '=') {
+          Parser.i++;
+          Parser.skipWhitespaces();
+          defaultValue = Parser.readNumber();
+          Parser.skipWhitespaces();
+        }
+
+        parameters.push({ name, type, optional, defaultValue });
+      } while (Parser.c === ',');
+
+    // Health checks
+    if (!(Parser.c === ')')) console.warn('Expected closing paranthesis!');
+    Parser.i++;
+    Parser.skipWhitespaces();
+    if (!(Parser.c === ';')) console.warn('Expected semicolon!');
+    Parser.i++;
+    Parser.skipWhitespaces();
+
+    methods.push(
+      `${isConstructor ? `constructor` : methodName}(${parameters
+        .map(
+          ({ name, type, optional, defaultValue }) =>
+            `${name}${optional ? '?' : ''}: ${
+              PrimitiveTypes.has(type) ? PrimitiveTypes.get(type) : type
+            }${defaultValue !== none ? ` = ${defaultValue}` : ''}`
+        )
+        .join(', ')}): ${
+        PrimitiveTypes.has(returnType)
+          ? PrimitiveTypes.get(returnType)
+          : returnType
+      };`
+    );
+  }
+
+  interfaces.push(
+    `export class ${interfaceName} extends EmObject {
+  ${methods.join('\n  ')}
+}`
+  );
+}
+
+const dts = `// Automatically generated by GDevelop.js/scripts/generate-dts.js
+
+class EmObject {
```

Comment:
nitpicking: 
```suggestion
class EmscriptenObject {
```

No reason not to be extra clear here, "Em" is not obvious for someone not familiar with Emscripten.